### PR TITLE
Remove some unnecessary error raising

### PR DESCRIPTION
**Pull request by GDYendell**

### CODE REVIEW

**Comment by GDYendell** [comment]
I think this will stop a lot of unnecessary errors client side. There is no need to raise an error for changing the process/rank because we are writing if the values are already set.

File: frameProcessor/src/FileWriterPlugin.cpp [hunk 1]
```diff
@@ -754,20 +755,37 @@ void FileWriterPlugin::configure(OdinData::IpcMessage& config, OdinData::IpcMess
...
+  // Check for rank number
   if (config.has_param(FileWriterPlugin::CONFIG_PROCESS_RANK)) {
-    this->concurrent_rank_ = config.get_param<size_t>(FileWriterPlugin::CONFIG_PROCESS_RANK);
-    LOG4CXX_DEBUG(logger_, "Process rank changed to " << this->concurrent_rank_);
+    size_t rank = config.get_param<size_t>(FileWriterPlugin::CONFIG_PROCESS_RANK);
+    if (this->concurrent_rank_ != rank) {
+      // If we are writing a file then we cannot change concurrent rank
+      if (this->writing_) {
+        LOG4CXX_ERROR(logger_, "Cannot change process rank whilst writing");
+        throw std::runtime_error("Cannot change process rank whilst writing");
```
Comment (MattTaylorDLS) [inline frameProcessor/src/FileWriterPlugin.cpp]: Is this caught somewhere? Do we definitely want to raise a runtime error?

File: frameProcessor/src/FileWriterPlugin.cpp [hunk 2]
```diff
@@ -754,20 +755,37 @@ void FileWriterPlugin::configure(OdinData::IpcMessage& config, OdinData::IpcMess
...
+  // Check for process number
   if (config.has_param(FileWriterPlugin::CONFIG_PROCESS_NUMBER)) {
-    this->concurrent_processes_ = config.get_param<size_t>(FileWriterPlugin::CONFIG_PROCESS_NUMBER);
-    LOG4CXX_DEBUG(logger_, "Concurrent processes changed to " << this->concurrent_processes_);
+    size_t processes = config.get_param<size_t>(FileWriterPlugin::CONFIG_PROCESS_NUMBER);
+    if (this->concurrent_processes_ != processes) {
+      // If we are writing a file then we cannot change concurrent processes
+      if (this->writing_) {
+        LOG4CXX_ERROR(logger_, "Cannot change concurrent processes whilst writing");
+        throw std::runtime_error("Cannot change concurrent processes whilst writing");
```
Comment (MattTaylorDLS) [inline frameProcessor/src/FileWriterPlugin.cpp]: Is this caught somewhere? Do we definitely want to raise a runtime error?

Comment (ajgdls) [inline frameProcessor/src/FileWriterPlugin.cpp]: Hi @MattTaylorDLS these configure errors I believe are caught and the appropriate error response is sent back to the client that requested the values.

Comment (MattTaylorDLS) [inline frameProcessor/src/FileWriterPlugin.cpp]: ok sounds good

File: frameProcessor/src/FileWriterPlugin.cpp (hunk 1 above)
Comment (GDYendell) [inline frameProcessor/src/FileWriterPlugin.cpp]: Yes this is caught. The configuration messages are all handled in a try catch block here: https://github.com/odin-detector/odin-data/blob/master/frameProcessor/src/FrameProcessorController.cpp#L100

File: frameProcessor/src/FileWriterPlugin.cpp (hunk 2 above)
Comment (GDYendell) [inline frameProcessor/src/FileWriterPlugin.cpp]: See above

**Comment by MattTaylorDLS** [comment]
Think this is all good to merge now yep?

**Comment by GDYendell** [comment]
I think so. It could do with testing as part of the DAQ stack before letting GDA have it, though.

**Comment by MattTaylorDLS** [comment]
Yes, agreed. We can merge it in now, and i'll test before updating to this in the work directory used by the live Eiger stack

**Pull request merged**